helm/prometheus-meta-operator/templates/prometheus-rules/prometheus-meta-operator.rules.yml

```
@@ -23,9 +23,9 @@ spec:
     - alert: "MatchingNumberOfPrometheusAndCluster"
       annotations:
         description: This alert is used to ensure we have as many workload cluster prometheus as we have workload cluster CR.
-      # This expression list all the cluster's that should exist using the cluster ID and compares them (using unless) to the running prometheus pods.
+      # This expression list all the cluster IDs that exist and are not being deleted and compares them (using unless) to the running prometheus pods.
       # If a prometheus is missing, this alert will fire. This alert will not check if a prometheus is running when it should not (e.g. deleted cluster)
-      expr: sum({__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status"}) by (cluster_id) unless sum(label_replace(kube_pod_container_status_running{container="prometheus", namespace=~".*-prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus"}, "cluster_id", "$2", "pod", "(prometheus-)(.+)(-.+)") ) by (cluster_id)
+      expr: sum({__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status", status!="Deleting"}}) by (cluster_id) unless sum(label_replace(kube_pod_container_status_running{container="prometheus", namespace=~".*-prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus"}, "cluster_id", "$2", "pod", "(prometheus-)(.+)(-.+)") ) by (cluster_id)
```

I tested it and we should actually fix it as:

```suggestion
      expr: (sum({__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status", status!="Deleting"}}) by (cluster_id) unless sum(label_replace(kube_pod_container_status_running{container="prometheus", namespace=~".*-prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus"}, "cluster_id", "$2", "pod", "(prometheus-)(.+)(-.+)") ) by (cluster_id)) > 0
```